skills/repo.go

```diff
@@ -61,3 +67,102 @@ func (r *Repository) AddType(t Type) error {
 	r.types[typeName] = t
 	return nil
 }
+
+// AllSkills returns all skills of the given type.
+// If skillType is the empty string, all skills are returned.
+func (r *Repository) AllSkills(skillType string) []*Skill {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	var result []*Skill
+	for _, skillsForSnap := range r.skills {
+		for _, skill := range skillsForSnap {
+			if skillType == "" || skill.Type == skillType {
+				result = append(result, skill)
+			}
+		}
+	}
+	sort.Sort(bySkillSnapAndName(result))
```

Comment:
The sorting here will feel slightly awkward, potentially. We can run with that for now and watch the use cases.